Amitlatestfaq (#9258)
harness/developer-hub: docs/continuous-delivery/deploy-srv-diff-platforms/kubernetes/kubernetes-faqs.md

https://developer.harness.io/docs/platform/connectors/cloud-providers/ref-cloud-providers/kubernetes-cluster-connector-settings-reference/#generate-the-service-account-token

> #### How to pass the manifest yaml to a shell step?
> 
> The manifest from previous step can be very big and if you pass this as input variable we will see failure like below:

Reviewer: I think you doubled up here
```suggestion
```

We have this avaialble as manifest expression so directly assign it within the step like below:

mymanifestvar=<+execution.steps.render_helm_chart_templates.k8s.manifestDryRun>

> #### Can we create resource in a differnt namespace than what is defined in the infrastructure?

Reviewer: ```suggestion
#### Can we create resources in different namespaces than what is defined in the infrastructure?
```

> It is not possible. All the resource being created as part of the deployment should be in the same namespace as defined in the infrastructure.

Reviewer: ```suggestion
It is not possible. All the resources being created as part of the deployment should be in the same namespace as defined in the infrastructure.
```

> #### Can I deploy two services with the same manifest and different values yaml?

Reviewer: ```suggestion
#### Can I deploy two services with the same manifest and different values.yaml files?
```

> Kuberenetes does not allow creating two resources with same name. Even if the service spec type is different still only one instance of the service can exist. For example if the same manifest is used to deploy a node port and a load balancer using different values.yaml, one iteration of the deployment will create only one instance of the resource.

Reviewer: ```suggestion
Kubernetes does not allow creating two resources with the same name. Even if the service spec type is different, only one instance of the service can exist. For example, if the same manifest is used to deploy a node port and a load balancer using a different values.yaml file, one iteration of the deployment will create only one instance of the resource.
```

If we use the same service to deploy a different service type it will overwrite the pre existing one as they have the same name.

> #### Is it possible to provide delegate selector for service hooks?

Reviewer: ```suggestion
#### Is it possible to provide a delegate selector for service hooks?
```

> We can not provide delegate selectors for service hook. They execute on the same delegate where the corresponding task is getting executed.

Reviewer: ```suggestion
We can not provide delegate selectors for a service hook. They execute on the same delegate where the corresponding task is getting executed.
```

> `mymanifestvar=<+execution.steps.render_helm_chart_templates.k8s.manifestDryRun>`

Reviewer: why is this here
